include/net/net_tls.h

```diff
@@ -0,0 +1,114 @@
+/** @file
+ * @brief Network TLS definitions
+ *
+ * An API for applications to configure TLS credentials.
+ */
+
+/*
+ * Copyright (c) 2018 Intel Corporation
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __NET_TLS_H
+#define __NET_TLS_H
+
+/**
+ * @brief Network TLS management
+ * @defgroup net_tls Network TLS management
+ * @ingroup networking
+ * @{
+ */
+
+#include <stddef.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** TLS credential type */
+enum net_tls_credential_type {
+	NET_TLS_CREDENTIAL_UNUSED,
+	NET_TLS_CREDENTIAL_CA_CERTIFICATE,
+	NET_TLS_CREDENTIAL_PSK,
+	NET_TLS_CREDENTIAL_PSK_ID
+};
+
+/** Secure tag, a reference for TLS credential */
+typedef int sec_tag_t;
+
+#if defined(CONFIG_NET_TLS) || defined(CONFIG_NET_DTLS)
+
+/** A list of secure tags that context should use */
+struct sec_tag_list {
```

Comment:
Following my comments on confused naming used with "sec_tag_list', my further grepping around makes me think that `struct sec_tag_list` is an internal structure and should moved to the internal header. (And something likely should be renamed still.)

Reply:
Yes, while I look at this now, it should be in the internal header.